src/redis_index.c

```diff
@@ -18,6 +18,26 @@
 
 RedisModuleType *InvertedIndexType;
 
+static inline void updateTime(SearchTime *searchTime, int32_t durationNS) {
+  if (RS_IsMock) return;
+
+  // 0 disables the timeout
+  if (durationNS == 0) {
+    durationNS = INT32_MAX;
+  }
+
+
+  struct timespec duration = { .tv_sec = durationNS / 1000,
+                               .tv_nsec = ((durationNS % 1000) * 1000000) };
+  clock_gettime(CLOCK_REALTIME_COARSE, &searchTime->current);
+
+  // The timeout mechanism is based on the monotonic clock, so we need another clock_gettime call
+  timespec monotoicNow = { .tv_sec = 0,
+                           .tv_nsec = 0 };
+  clock_gettime(CLOCK_MONOTONIC_RAW, &monotoicNow);
+  rs_timeradd(&monotoicNow, &duration, &searchTime->timeout);
```

Comment:
Do we always want to update these together?
If not consider splitting the API

Reply:
I think we do want to always update them together.
They both relate to the current point in time, just with different clocks.

Reply:
Seems like a bit of a mixture.. but OK